composer

```diff
@@ -280,13 +274,40 @@ if (!class_exists('ComposerWrapper')) {
 
             $this->verifyChecksum($installerPathName);
 
-            \passthru(
-                \sprintf(
-                    '%s %s --install-dir=%s',
-                    \escapeshellarg($this->getPhpBinary()),
+            $majorVersion = $this->params->getForceMajorVersion();
+            $majorVersionFlag = null;
+            if (null !== $majorVersion) {
+                $helpCommandParts = array(
                     \escapeshellarg($installerPathName),
-                    \escapeshellarg($dir)
-                ),
+                    '--no-ansi',
+                    '--help',
+                );
+                $helpCommand = implode(' ', $helpCommandParts);
+                if ($this->supportsForceVersionFlag($helpCommand, $majorVersion)) {
+                    $majorVersionFlag = '--' . $majorVersion;
+                } else {
+                    $this->showError(
+                        "Installer doesn't allow to specify major version.\n" .
```

Comment:
@kamazee can you provide here some details, just for my information. When this situation possible? I've checked https://getcomposer.org/installer and it still supports major version specification

Reply:
I want wrapper to continue working with a warning if the installer drops support for this flags (I assume that options come and go; nobody guarantees low long they will be there)